src/metabase/models/search_index_metadata.clj

```diff
@@ -0,0 +1,74 @@
+(ns metabase.models.search-index-metadata
+  (:require
+   [metabase.models.interface :as mi]
+   [methodical.core :as methodical]
+   [toucan2.core :as t2]))
+
+(methodical/defmethod t2/table-name :model/SearchIndexMetadata [_model] :search_index_metadata)
```

Comment:
This name feels too long, but I'm a bit nervous about reusing the `:search_index` name again with a different meaning.